docs/proposals/rollout/README.md

```diff
@@ -0,0 +1,308 @@
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story #1: Deployment Rolling Update with 2 maxUnavailable and 1 maxSurge
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: rollout-demo
+spec:
+  replicas: 10
+  strategy:
+    rollingUpdate:
+      maxUnavailable: 2
+      maxSurge: 1
+  template:
+    metadata:
+      labels:
+        app: rollout-demo
+    spec:
+      containers:
+        - name: main
+          image: nginx:v2 # upgrade rollout-demo from V1 -> V2
+  selector:
+    matchLabels:
+      app: rollout-demo
+---
+apiVersion: karmada.io/v1alpha2
+kind: ResourceBinding
+metadata:
+  name: rollout-demo-deployment
+spec:
+  clusters: # scheduled to 2 clusters
+    - name: cluster-1
+      replicas: 4
+    - name: cluster-2
+      replicas: 6
+```
+
+We expect that the Deployment Rolling Update with 2 maxUnavailable and 1 maxSurge will be scheduled to 2 clusters, and the rollout process will be consistent with the semantics in a single-cluster setting.
+During rolling processing, the number of unavailable Pods will not exceed 2, and the number of Pods will not exceed 11.
+
+#### Story #2: Batch Release with Partition
+```yaml
+apiVersion: apps.kruise.io/v1alpha1
+kind: CloneSet
+metadata:
+  name: rollout-demo
+spec:
+  replicas: 10
+  updateStrategy:
+    partition: 4
+    maxUnavailable: 2
+  template:
+    metadata:
+      labels:
+        app: rollout-demo
+    spec:
+      containers:
+        - name: main
+          image: nginx:v2 # upgrade rollout-demo from V1 -> V2
+  selector:
+    matchLabels:
+      app: rollout-demo
+---
+apiVersion: karmada.io/v1alpha2
+kind: ResourceBinding
+metadata:
+  name: rollout-demo-cloneset
+spec:
+  clusters: # scheduled to 2 clusters
+    - name: cluster-1
+      replicas: 3
+    - name: cluster-2
+      replicas: 7
+```
+
+
+We expect that the CloneSet cloud be paused when the number of Pods with the new image reaches 4, and the rollout process will be consistent with the semantics in a single-cluster setting. 
+During rolling processing, the number of unavailable Pods will not exceed 2.
+
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? 
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+- May further reduce the efficiency of distribution synchronization of workloads.
+
+## Architecture
+The primary approach involves using feedback from the StatusController on single-cluster workload statuses to derive the next rollout configurations. This ensures that throughout the Rollout process, rollout configurations such as `Partition`, `MaxUnavailable`, and `MaxSurge` maintain their original native meanings, and that the federated Rollout behavior aligns with single-cluster  practices.
+
+![architecture](rollout-mechanism-arch.png)
```

Comment:
It would be great if we could have more detailed instructions to demonstrate the process.